File: c++/a647.cpp
#include <stdio.h>
const double cap=1e-5;

int main()
{
    int t;
    scanf("%d",&t);
    while (t--)
    {
        int m,p;
        scanf("%d%d",&m,&p);
        double rat=(double)(p-m)/m*100;
        if (rat>=0.0) printf("%.2lf%% ",rat+cap);
        else printf("%.2lf%% ",rat-cap);
        if (rat >= 10.0 || rat <= -7.0) printf("dispose\n");
        else printf("keep\n");
    }
}

File: python/a647.py
import sys

for num in sys.stdin:
    num=int(num.replace('\n',''))
    for i in range(num):
        m,p=list(map(int,input().split(' ')))
        x=((p/m)-1)*100
        x=round(x*100)/100.0
        if -7.00 >= x or x>= 10.00:
            print ('%.2f'%x+'% dispose')
        else:
            print ('%.2f'%x+'% keep')